spring/boot-webflux-autoconfigure/src/main/java/com/linecorp/armeria/spring/web/reactive/ArmeriaReactiveWebServerFactory.java

```diff
@@ -0,0 +1,352 @@
+            final String[] enabledProtocols = ssl.getEnabledProtocols();
+            if (enabledProtocols != null) {
+                sslBuilder.protocols(Arrays.copyOf(enabledProtocols, enabledProtocols.length));
+            }
+
+            final String[] ciphers = ssl.getCiphers();
+            if (ciphers != null) {
+                sslBuilder.ciphers(ImmutableList.copyOf(ciphers));
+            }
+
+            final ClientAuth clientAuth = ssl.getClientAuth();
+            if (clientAuth != null) {
+                switch (clientAuth) {
+                    case NEED:
+                        sslBuilder.clientAuth(io.netty.handler.ssl.ClientAuth.REQUIRE);
+                        break;
+                    case WANT:
+                        sslBuilder.clientAuth(io.netty.handler.ssl.ClientAuth.OPTIONAL);
+                        break;
+                }
+            }
+
+            sb.tls(sslBuilder.build());
+        } catch (Exception e) {
+            throw new IllegalStateException("Failed to configure TLS: " + e);
+        }
+    }
+
+    private static KeyManagerFactory getKeyManagerFactory(
+            Ssl ssl, @Nullable SslStoreProvider sslStoreProvider) throws Exception {
+        final KeyStore store;
+        if (sslStoreProvider != null) {
+            store = sslStoreProvider.getKeyStore();
+        } else {
+            store = loadKeyStore(ssl.getKeyStoreType(), ssl.getKeyStore(), ssl.getKeyStorePassword());
+        }
+
+        final KeyManagerFactory keyManagerFactory =
+                KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+
+        String keyPassword = ssl.getKeyPassword();
+        if (keyPassword == null) {
+            keyPassword = ssl.getKeyStorePassword();
+        }
+
+        keyManagerFactory.init(store, keyPassword != null ? keyPassword.toCharArray()
+                                                          : null);
+        return keyManagerFactory;
+    }
+
+    private static TrustManagerFactory getTrustManagerFactory(
+            Ssl ssl, @Nullable SslStoreProvider sslStoreProvider) throws Exception {
+        final KeyStore store;
+        if (sslStoreProvider != null) {
+            store = sslStoreProvider.getTrustStore();
+        } else {
+            store = loadKeyStore(ssl.getTrustStoreType(), ssl.getTrustStore(), ssl.getTrustStorePassword());
+        }
+
+        final TrustManagerFactory trustManagerFactory =
+                TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+        trustManagerFactory.init(store);
+        return trustManagerFactory;
+    }
+
+    @Nullable
+    private static KeyStore loadKeyStore(
+            @Nullable String type,
+            @Nullable String resource,
+            @Nullable String password) throws IOException, GeneralSecurityException {
+        if (resource == null) {
+            return null;
+        }
+        final KeyStore store = KeyStore.getInstance(firstNonNull(type, "JKS"));
```

Comment:
It seems that making Spring user confused, because they are familiar with the current way which is based on keystore. So it would be better to improve this later, by extending `ArmeriaSettings`.